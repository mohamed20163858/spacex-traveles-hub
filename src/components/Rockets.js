import { useSelector } from 'react-redux';
import RocketCard from './rocketCard';

const Rockets = () => {
  const infoUI = [];
  const infoState = useSelector((state) => state.rocketInfo);
  for (let i = 0; i < infoState.length; i += 1) {
    infoUI.push(<RocketCard
      rocketId={infoState[i].id}
      rocketName={infoState[i].rocket_name}
      rocketDescription={infoState[i].description}
      rocketImage={infoState[i].flickr_images[0]}
      reserved={Boolean(infoState[i].reserved)}
      key={`rocket-${infoState[i].id}`}
    />);
  }
  return (
    <div data-testid="momo-id">
      {infoUI}
    </div>
  );
};
export default Rockets;
