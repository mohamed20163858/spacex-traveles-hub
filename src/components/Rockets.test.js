import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import Rockets from './rocketCard';

function TestRockets() {
  return (
    <Provider store={store}>
      <Rockets />
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestRockets />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('simulate reserve a rocket ', () => {
  // ARRANGE
  render(<TestRockets />);

  // ACT
  userEvent.click(screen.getByText('Reserve Rocket'));
  // Arrange
  const result = screen.getByTestId('rocket-reserved');
  // ASSERT
  expect(result).toBeInTheDocument();
});
