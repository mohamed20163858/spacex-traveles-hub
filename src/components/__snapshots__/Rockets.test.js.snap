// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rocket-card"
>
  <img
    alt="rocket"
    width="200"
  />
  <div
    className="card-info"
  >
    <h2 />
    <p>
      <span
        className=""
        data-testid=""
      />
    </p>
    <button
      className="unreserved"
      onClick={[Function]}
      type="submit"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
